{"project": "pandas", "bug": 63, "project_url": "https://github.com/pandas-dev/pandas", "file_changed": "pandas/core/indexing.py", "buggy_commit_id": "e5c65bf", "fixed_commit_id": "e1ca66bae38b8026079dfcbe0edad5f278546608", "lines_deleted": {"2089": "        for ax, i in zip(self.obj.axes, key):\n", "2090": "            if ax.is_integer():\n", "2091": "                if not is_integer(i):\n", "2092": "                    raise ValueError(\n", "2093": "                        \"At based indexing on an integer index \"\n", "2094": "                        \"can only have integer indexers\"\n", "2095": "                    )\n", "2096": "            else:\n", "2097": "                if is_integer(i) and not (ax.holds_integer() or ax.is_floating()):\n", "2098": "                    raise ValueError(\n", "2099": "                        \"At based indexing on an non-integer \"\n", "2100": "                        \"index can only have non-integer \"\n", "2101": "                        \"indexers\"\n", "2102": "                    )\n", "2103": "        return key\n"}, "lines_added": {"2089": "        lkey = list(key)\n", "2090": "        for n, (ax, i) in enumerate(zip(self.obj.axes, key)):\n", "2091": "            lkey[n] = ax._convert_scalar_indexer(i, kind=\"loc\")\n", "2092": "\n", "2093": "        return tuple(lkey)\n"}}